src/bin/jcli_app/utils/output_format.rs

```diff
@@ -0,0 +1,33 @@
+use serde::Serialize;
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+pub struct OutputFormat {
+    /// Format of output data. Possible values: json, yaml.
+    /// Any other value is treated as a custom format using values from output data structure.
+    /// Syntax example: "{node_id}: up for {stats.uptime}".
+    /// Full syntax reference: https://docs.rs/tinytemplate/1.0.2/tinytemplate/syntax/index.html
+    #[structopt(long = "output-format", default_value = "yaml")]
+    format: String,
+}
```

Comment:
For better type safety this could be:

```rest
pub enum OutputFormat {
    Json,
    Yaml,
    Template(String),
}
```

You'll need a custom `StructOpt` impl for it, though.

Reply:
The `format` field is private and used in only one place, which wouldn't benefit from enum anyway. Isn't it an overkill?

Reply:
Then you can match the output modes in a type-safe rather than stringized way.